src/cli/self_update/windows.rs

```diff
@@ -460,35 +463,39 @@ pub(crate) fn do_add_to_path(process: &Process) -> Result<()> {
     do_add_to_programs(process)
 }
 
-fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
+fn _apply_new_path(new_path: Option<String>) -> Result<()> {
```

Comment:
Looks like this could be an `Option<&str>` (in the sense that the function body doesn't need to take/pass ownership)?

Reply:
In this function there is a code snippet which verifies whether this string is empty. Because this `is_empty()` is implemented for `String`, so I chose `String`.

Reply:
is_empty() is actually defined on &str and can be used via String's Deref<Target = &str> implementation.

Reply:
> is_empty() is actually defined on &str and can be used via String's Deref<Target = &str> implementation.

Oh OK, got it!

Reply:
I attempted using `&str` but I met lifetime problems😅, so I continue using `String` unless I resolved these problems.